### ISSUE TITLE
Add options to remove paths if invalid and to update timestamps if valid

### ISSUE THREAD

**Issue by ddriddle** [techservicesillinois/cache-validation#7]
Currently, I have the following code which deletes a path if it is invalid or updates the timestamp if the path is valid:

```yaml
    - name: Validate tox virtualenv
      uses: techservicesillinois/cache-validation@v1
      id: tox
      with:
        path: .tox
        cache_hit: ${{ steps.ghcache.outputs.cache-hit }}

    - name: Validate wheel cache
      uses: techservicesillinois/cache-validation@v1
      id: cache
      with:
        path: cache
        cache_hit: ${{ steps.ghcache.outputs.cache-hit }}

    # rm tox env so make will rebuild if invalid
    - name: Remove tox virtualenv
      if: steps.tox.outputs.valid != 'true'
      run: rm -rf .tox
      shell: bash

    # touch cache so make will rebuild it if invalid
    - name: Remove wheel cache
      if: steps.cache.outputs.valid != 'true'
      run: rm -rf cache
      shell: bash

    # touch tox env so make will not rebuild it if valid
    - name: Update tox timestamps
      if: steps.tox.outputs.valid == 'true'
      run: touch -c .tox/wheel/pyvenv.cfg
      shell: bash

    # touch cache so make will not rebuild it if valid
    - name: Update wheel timestamps
      if: steps.cache.outputs.valid == 'true'
      run: touch -c cache
      shell: bash
```

To reduce the amount of repeated code, I purpose adding the options `remove_invalid_paths`, `remove_invalid_files`, and `touch`:

```yaml
  remove_invalid_paths:    
    description: 'A boolean value that when set to true causes a path to be recursively removed if it contains invalid files.'
    required: false
  remove_invalid_files:    
    description: 'A boolean value that when set to true causes invalid files in a path to be removed, valid files are always retained.'
    required: false
  touch:    
    description: 'A boolean value that when set to true recursively updates the modification and access times of each path.'
    required: false
```

The code above with the options purposed would reduce to:

```yaml
    - name: Validate tox virtualenv
      uses: techservicesillinois/cache-validation@v1
      id: tox
      with:
        path: .tox
        cache_hit: ${{ steps.ghcache.outputs.cache-hit }}
        remove_invalid_paths: true
        touch: true

    - name: Validate wheel cache
      uses: techservicesillinois/cache-validation@v1
      id: cache
      with:
        path: cache
        cache_hit: ${{ steps.ghcache.outputs.cache-hit }}
        remove_invalid_files: true
        touch: true
```

**Comment by edthedev** [comment]
Prioritizing this as it will simplify #6

**Comment by ddriddle** [comment]
Closed by #11, #12, and #14.
